annotate.el

```diff
@@ -595,11 +595,23 @@ specified by `from' and `to'."
                   (when delete-enclosed
                     (annotate-delete-chains-in-region chain-end region-stop))))
               (annotate-line (eol)
-                (let* ((bol (annotate-beginning-of-line-pos)))
-                  (goto-char bol)
-                  (set-mark (point))
-                  (goto-char eol)
-                  (annotate-annotate))))
+                (let* ((bol                     (annotate-beginning-of-line-pos))
```

Comment:
These sub-functions are easy enough to follow, but on the whole `annotate-annotate` is becoming a bit too long for my liking. A few comments would improve readability a lot, I feel, or even breaking up into multiple functions.

Reply:
Hi @bastibe !!!

You are right, there is no reason (except laziness ;-) ) that prevents adding documentation to these functions. As far as i know local functions does not supports docstring so i have added documentation as mere comments. I hope this is OK.

Bye!
C.